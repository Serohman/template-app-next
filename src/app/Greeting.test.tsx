import "@testing-library/jest-dom";
import {render, screen} from "@testing-library/react";
import React from "react";
import logo from "../logo.png";
import Greeting from "./Greeting";
import styles from "./Greeting.module.css";

// Mock the CSS module and image import
jest.mock("./Greeting.module.css", () => ({
  title: "mocked-title-class",
}));
jest.mock("../logo.png", () => "mocked-logo.png");

describe("Greeting", () => {
  it("renders the correct greeting", () => {
    render(<Greeting name="World" />);
    expect(screen.getByText("Hello, World!")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const {container} = render(<Greeting name="Snapshot" />);
    expect(container).toMatchSnapshot();
  });

  it("uses mocked CSS module and image", () => {
    expect(styles.title).toBe("mocked-title-class");
    expect(logo).toBe("mocked-logo.png");
  });

  it("calls a function with jest.spyOn", () => {
    const logSpy = jest.spyOn(console, "log").mockImplementation(() => {});
    console.log("Test log");
    expect(logSpy).toHaveBeenCalledWith("Test log");
    logSpy.mockRestore();
  });
});
