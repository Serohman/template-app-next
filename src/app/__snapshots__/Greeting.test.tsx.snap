// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting matches snapshot 1`] = `
<div>
  <h1>
    Hello, 
    Snapshot
    !
  </h1>
</div>
`;
